runtime/lua/vim/filetype.lua

```diff
@@ -1424,6 +1424,8 @@ local pattern = {
       return "git"
     end
   end,
+  -- Neovim only
+  [".*/queries/.*%.scm"] = "query", -- tree-sitter queries
```

Comment:
"query" is a very generic name, which is hinted by the presence of the `tree-sitter queries` comment (which users won't see... and thus will have no idea what `ft=query` means). 

Should we not use a common prefix like `treesitter-query` ?

Reply:
Is there precedent for that? Other filetypes also call themselves whatever they wish, with no respect for others... "Query" is what upstream calls them, at least.

Reply:
(Meaning, this is not something I made up on the spot for this PR.)

Reply:
The question answers itself: in every UI, documentation, and discussion about this filetype, it will always need to be disambiguated. No one knows what "query" refers to, it's too generic.

What happens when this kind of choice is avoided by the software developer: everyone pays a tiny price forever, because the software developer avoided the choice once. The one-time choice is to think of an unambiguous name such as `tsquery`.

Reply:
Yes. And yet, here we are. My point is that this is not a decision that can be made in isolation here; I was just following the choice already made.

Reply:
We have to make the decision or we're just foisting it on users. It's like if they named the format "text".

> Is there precedent for that? 

CMakeLists.txt has a `.txt` extension but we don't named its filetype `text`.

This also has a discoverability cost: "query" is hard to search for. And if there are other treesitter-related formats we want to add, using a common prefix helps users find all such related syntaxes.

This applies to naming in pretty much all contexts across a programming project.

If I have to type out this much every time we have a naming question, that doesn't scale. We need to use this judgement in all our naming choices.

Reply:
But, again, this is a filetype that is already used in the wild. If we set these files to a _different_ type, this would defeat the whole purpose of doing it here.

I'm fine with `tsquery` and will happily make a follow-up PR if you can get upstream to change as well.